### ISSUE TITLE
`PhantomData<T>` serializes to unit struct, not `T`

### ISSUE THREAD

**Issue by HactarCE** [nicolaiunrein/zod#5]
The crate docs says that `PhantomData<T>` currently translates to the Zod schema `T`. This should not be the case; `PhantomData<T>` does not contain a `T` and is thus [serialized as a unit struct](https://docs.rs/serde/latest/src/serde/ser/impls.rs.html#119-124). It merely appears to contain a a `T` for the sake of subtyping/variance, which I believe is irrelevant for this crate.

**Comment by nicolaiunrein** [comment]
I agree. However I am almost done with a complete rewrite to fully support generics. I'll keep this issue in mind.
